pkg/cmd/pipeline/sign.go

```diff
@@ -70,7 +73,13 @@ or using kms
 				return err
 			}
 
-			crd := &v1beta1.Pipeline{}
+			var crd metav1.Object
```

Comment:
We are reading the file provided by user as input, we can get the info about version from file content. Why we need to take the input from user?

Also --pipeline-version flag does not look make sense for task commands

Reply:
naming is hard 😅 I think the idea is to convey which tektoncd/pipeline version we want... 

but yeah that may be confusing?

any other suggestions? --crd-version perhaps ?

Reply:
Thanks for the suggestion, but I don't know what would be the best way to do this. 
How to read the version info from the file content? Scan the string? How to get the version `v1` for this case?
e.g.
```yaml
# apiVersion: v1beta1
apiVersion: v1
...
```

Reply:
--pipeline-version means (Tekton) Pipeline Version, it was suggested in the previous comment of this PR. Maybe rename it to --api-version?Wdyt?

Reply:
Replied here https://github.com/tektoncd/cli/pull/2216#issuecomment-2182139865

It does omit the commented one than in file